### ISSUE TITLE
Validation error when use a custom attribute in fhir resource in first-class attribute by fhir endpoint should be extra property

### ISSUE THREAD

**Issue by jyeros** [Aidbox/Issues#146]
When I try to post a single resource to a custom attribute with `isCollection=true` and type reference if post a single resource instead an array the response is: `expected an array` how it should be.
But if I try to post an array with a reference by fhir endpoint with `reference: ResourceType/id` it works, but no translate it when use aidbox endpoint. I think it is related with this issue https://github.com/Aidbox/Issues/issues/135

For example

```
POST /Entity

type: resource
module: mymodule
id: MyCustomEntity
resourceType: Entity
```

```
POST /Attribute

description: Some description
path: [owner]
refers: [Practitioner, PractitionerRole, Organization, Patient]
type: {id: Reference, resourceType: Entity}
resourceType: Attribute
module: mymodule
id: MyCustomEntity.owner
resource: {id: MyCustomEntity, resourceType: Entity}
isRequired: true
isCollection: true
```
```
POST /fhir/MyCustomEntity

owner:
  - resource: Patient/pat1
```
```
Get /MyCustomEntity/ec85c392-fb4d-4187-9888-6c539d1443da
---
owner:
- {resource: Patient/pat1}
id: ec85c392-fb4d-4187-9888-6c539d1443da
resourceType: MyCustomEntity
meta: {lastUpdated: '2019-08-29T14:44:47.027Z', versionId: '1886'}
```
```
---
Get /fhir/MyCustomEntity/ec85c392-fb4d-4187-9888-6c539d1443da
id: ec85c392-fb4d-4187-9888-6c539d1443da
meta: {lastUpdated: '2019-08-29T14:44:47.027Z', versionId: '1886'}
owner:
- {resource: Patient/pat1}
resourceType: MyCustomEntity
```

The reference is not translate to aidbox reference.

**Comment by niquola** [comment]
```
POST /fhir/MyCustomEntity

owner:
  - resource: Patient/pat1
```

should be 

```
POST /fhir/MyCustomEntity

owner:
  - reference: Patient/pat1

```

**Comment by jyeros** [comment]
Sorry, my bad is reference, and i try this example and works but when i try a custom attribute in a class that exist doesn't work, let me sent you the example

**Comment by jyeros** [comment]
```
Post /Attribute

{
  "path": [
    "temp"
  ],
  "module": "phyl2",
  "description": "temp",
  "id": "CatalogEntry.temp",
  "resource": {
    "id": "CatalogEntry",
    "resourceType": "Entity"
  },
  "type": {
    "id": "Reference",
    "resourceType": "Entity"
  },
  "refers": [
    "Organization"
  ],
  "resourceType": "Attribute",
  "isCollection": true
}
```
```
Post /fhir/CatalogEntry

{
    "type": {
        "text": "service"
    },
    "status": "active",
    "orderable": false,
    "identifier": [
        {
            "use": "official",
            "value": "34",
            "system": "http://test.test"
        }
    ],
    "referencedItem": {
        "reference": "ObservationDefinition/bbc4c7f8-a9c9-423c-a359-81c7d412ba91"
    },
    "resourceType": "CatalogEntry",
    "temp":[
        {
            "reference": "Organization/620cef1c-1e63-4b69-9d49-52a024eaa141"
        }
    ]
}
```
The response is 
```
resourceType: OperationOutcome
text: {status: generated, div: Invalid resource}
issue:
- {severity: fatal, code: invalid, expression: CatalogEntry.temp.0.reference, diagnostics: extra property}
```

**Comment by jyeros** [comment]
It works as excepted in a custom resource, the problem is when a try to create a custom attribute to a fhir resource and use it in first-class attribute by fhir endpoint, I should do it by aidbox endpoint instead fhir endpoint. Maybe the error here should be `temp` is an extra property even if the attribute is created

**Comment by niquola** [comment]
Fixed in 0.4.7-SNAPSHOT
